### PR TITLE
Refactor to be based on puppet releases not OS distributions

### PR DESCRIPTION
**Pull request by gregswift**
- Remove centos6, centos7, and fedora20
- Create puppet3 and puppet4 image targets based on Fedora21 (We dont use Fedora22 because puppet isn't compatible with Ruby 2.2.2 apparently)
- We use 2 separate dockerfiles despite the only difference being a single variable due to docker build not accepting variables
- Restructure the Dockerfiles to reduce layers
- Use a gemfile to manage the gem installs in the image
- Add atomic app compatible LABEL for RUN as an example

### CODE REVIEW

**Comment by teriyakichild** [comment]
This was tested on Fedora 21.  Running `make` successfully built two images:

rspec-puppet             puppet4             5d4d02c4c91a        36 minutes ago      806.9 MB 
rspec-puppet             puppet3             306ff47a62e3        39 minutes ago      807.4 MB

**Comment by teriyakichild** [comment]
Where are the steps to actually test a puppet module?

**Comment by gregswift** [comment]
@teriyakichild the main usage docs would be in dockerhub, and I need to update those. But I updated the README with some usage just for you :)

File: README.md [hunk 1]
```diff
@@ -2,3 +2,55 @@ docker-rspec-puppet
...
+
+Usage
+=====
+The simplest docker run for this module is, which must be done from inside a puppet module.
+
+    docker run -t -v \$(pwd):/module:z gregswift/rspec-puppet:latest
+
+If your puppet module's fixtures include some git over ssh repositories you can do this
+
+    docker run -t --name NAME -v $(pwd):/module:z -v ${SSH_AUTH_SOCK}:/ssh-agent:z -e SSH_AUTH_SOCK=/ssh-agent gregswift/rspec-puppet:latest
```
Comment (teriyakichild) [inline README.md]: Remove the "--name" parameter. 
Add the "--rm" parameter.

**Comment by teriyakichild** [comment]
You should be able to squash some of these commits.

**Pull request merged**